modules/common-fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/common/fs2/Run.scala

```diff
@@ -0,0 +1,152 @@
+                      val env = Environment
+                        .make[F, Array[Byte]](
+                          blocker,
+                          ec,
+                          parsed,
+                          Source.filesystem[F](blocker, p.dir),
+                          goodSink,
+                          piiSink,
+                          badSink,
+                          _.flatMap(_ => Stream.eval(Sync[F].unit)),
+                          identity,
+                          processor
+                        )
+                      runEnvironment[F, Array[Byte]](env, false)
+                    case _ =>
+                      val env = Environment
+                        .make[F, A](
+                          blocker,
+                          ec,
+                          parsed,
+                          mkSource(blocker, file.auth, file.input),
+                          goodSink,
+                          piiSink,
+                          badSink,
+                          checkpointer,
+                          getPayload,
+                          processor
+                        )
+                      runEnvironment[F, A](env, ordered)
+                  }
+              } yield exit
+            }
+        ).flatten
+      case Left(error) =>
+        Sync[F].delay(System.err.println(error)) >> Sync[F].pure(ExitCode.Error)
+    }
+
+  private def initAttributedSink[F[_]: Concurrent: ContextShift: Timer](
+    blocker: Blocker,
+    auth: Authentication,
+    output: Output,
+    mkGoodSink: (Blocker, Authentication, Output) => Resource[F, AttributedByteSink[F]],
+  ): Resource[F, AttributedByteSink[F]] =
+    output match {
+      case Output.FileSystem(path) =>
+        Sink.fileSink[F](path, blocker).map(sink => row => sink(row.data))
+      case _ =>
+        mkGoodSink(blocker, auth, output)
+    }
+
+  private def runEnvironment[F[_]: ConcurrentEffect: ContextShift: Parallel: Timer, A](
+    environment: Resource[F, Environment[F, A]],
+    ordered: Boolean
+  ): F[ExitCode] =
+    environment.use { env =>
+      val log = Logger[F].info("Running enrichment stream")
+      val enrich = Enrich.run[F, A](env, ordered)
+      val updates = Assets.run[F, A](env)
+      val reporting = env.metrics.report
+      val flow = enrich.merge(updates).merge(reporting)
+      log >> flow.compile.drain.attempt.flatMap {
+        case Left(exception) =>
+          sendToSentry(exception, env.sentry) >>
+          Logger[F].error(s"The Enrich job has stopped") >>
+          Sync[F].raiseError[ExitCode](exception)
+        case Right(_) =>
+          Sync[F].pure(ExitCode.Success)
+      }
+    }
+
+  private def sendToSentry[F[_]: Sync](error: Throwable, sentry: Option[SentryClient]): F[Unit] =
+    sentry match {
+      case Some(client) =>
+        Sync[F].delay(client.sendException(error)) >> Logger[F].info("Sentry report has been sent")
+      case None =>
+        Sync[F].unit
+    }
```

Comment:
@chuwy was there a specific reason why it was performed outside `F` in [previous implementation](https://github.com/snowplow/enrich/blob/master/modules/pubsub/src/main/scala/com/snowplowanalytics/snowplow/enrich/pubsub/Main.scala#L80-L87) ?

Reply:
Hard to say. I was just afraid that if something really bad happens - the CE runtime could be broken, so `IO`s wouldn't be executed anymore. But that's a fairly dumb assumption - I cannot explain why/how it would happen, so agree to leave it the way you did.